master/internal/usergroup/api_groups.go

```diff
@@ -30,6 +32,15 @@ func (a *UserGroupAPIServer) CreateGroup(ctx context.Context, req *apiv1.CreateG
 		err = apiutils.MapAndFilterErrors(err)
```

Comment:
does this ```mapAndFilterErrors``` here make a deny case just return ```Internal Service Error```

Reply:
yeah if it doesn't pass the filter, it returns an internal error

Reply:
We should find a way to not filter out these access deny errors

Reply:
Hm, good point. `MapAndFilterErrors` was intended to interact with DB sentinel errors, but I agree that it would be more useful if either:
- it blanket passed through all GRPC errors. We could perhaps use [`status.Code(err)`](https://pkg.go.dev/google.golang.org/grpc/status#Code) to detect for these and pass them through.
- its scope was broadened to be a standardization layer for all API-level errors
- it only stopped errors that came from the DB-related calls (e.g. stuff from `database/sql`, Bun, sqlx, etc.)